sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -161,7 +161,11 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
       val filterSet = ExpressionSet(filters)
 
       val normalizedFilters = DataSourceStrategy.normalizeExprs(
-        filters.filter(_.deterministic), l.output)
+        filters.filter { f =>
+          f.deterministic &&
+            !SubqueryExpression.hasSubquery(f) &&
+            !f.exists(_.isInstanceOf[PythonUDF])
```

Comment:
`FileSourceStrategy` is planning phase, I think PythonUDF should be collected and put in special operators already?